src/transforms/collapse-intersection-interfaces-transform.ts

```diff
@@ -32,11 +32,11 @@ export function collapseIntersectionInterfacesTransformFactoryFactory(
 
             function visitTypeAliasDeclaration(node: ts.TypeAliasDeclaration) {
                 if (
-                    helpers.isIntersectionTypeNode(node.type)
-                    && node.type.types.every((type) => helpers.isTypeLiteralNode(type))
+                    ts.isIntersectionTypeNode(node.type)
+                    && node.type.types.every((type) => ts.isTypeLiteralNode(type))
                 ) {
-                    const allMembers = node.type.types
-                        .map((type: ts.TypeLiteralNode) => type.members)
+                    const allMembers = (node.type.types as ts.NodeArray<ts.TypeLiteralNode>)
```

Comment:
Why we have to cast to `ts.NodeArray<ts.TypeLiteralNode>` here?

Reply:
Because TypeScript can't figure out the previous `node.type.types.every(...)`, and still think the type of `node.type.types` is `ts.NodeArray<ts.TypeNode>`, rather than `ts.NodeArray<ts. TypeLiteralNode>`.

![image](https://user-images.githubusercontent.com/6076919/35180610-dc27425c-fdee-11e7-8979-a0a09d3b0c68.png)

I can add a comment here.

Reply:
ok I think casting should be fine for now. I'll dig deeper into it soon. 


/cc @DanielRosenwasser